### PR TITLE
util: CBufferedFile fixes

### PR DESCRIPTION
**Pull request by LarryRuane**
Cherry-pick https://github.com/bitcoin/bitcoin/pull/16577

### CODE REVIEW

File: src/test/streams_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+    boost::filesystem::remove("streams_test_tmp");
+}
+
+BOOST_AUTO_TEST_CASE(streams_buffered_file_rand)
+{
+    // Make this test deterministic.
+    seed_insecure_rand(true);
+
+    for (int rep = 0; rep < 500; ++rep) {
```
Comment (str4d) [inline src/test/streams_tests.cpp]: This uses 50 iterations in the upstream PR, but if this isn't too slow then I'm fine with more iterations here.

File: src/streams.h [hunk 2]
```diff
@@ -531,19 +531,20 @@ class CBufferedFile
             readNow = nAvail;
         if (readNow == 0)
             return false;
-        size_t read = fread((void*)&vchBuf[pos], 1, readNow, src);
-        if (read == 0) {
+        size_t nBytes = fread((void*)&vchBuf[pos], 1, readNow, src);
+        if (nBytes == 0) {
```
Comment (str4d) [inline src/streams.h]: These changes are in this PR because we haven't cherry-picked bitcoin/bitcoin#8808.

File: src/test/streams_tests.cpp (hunk 1 above)
Comment (LarryRuane) [reply src/test/streams_tests.cpp]: This was an oversight on my part; I wanted to do a little extra testing and forgot to change it back to 50. The time (both CPU and real) to run this test is about 1.4s with 50 iterations, and 1.7s with 500 iterations. It's probably worth the 0.3 seconds to get better test coverage, so I'll leave it at 500.

**Comment by str4d** [comment]
Timing out on @daira's review request.

@zkbot r+

File: src/test/streams_tests.cpp [hunk 3]
```diff
@@ -0,0 +1,256 @@
+// Copyright (c) 2019 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php.
+
+#include "main.h"
+#include "test/test_bitcoin.h"
```
Comment (str4d) [inline src/test/streams_tests.cpp]:
```suggestion
#include "test/test_bitcoin.h"
#include "test/test_random.h"
```

**Comment by str4d** [comment]
@zkbot r+

**Pull request merged**